packages/dev/core/src/Materials/Textures/baseTexture.ts

```diff
@@ -319,9 +321,17 @@ export class BaseTexture extends ThinTexture implements IAnimatable {
         return this._texture != null && this._texture._isRGBD;
     }
     public set isRGBD(value: boolean) {
+        if (value === this.isRGBD) {
+            return;
+        }
+
         if (this._texture) {
             this._texture._isRGBD = value;
         }
+
+        this.getScene()?.markAllMaterialsAsDirty(Constants.MATERIAL_TextureDirtyFlag, (mat) => {

```

Comment:
Maybe this should be inside the `if` above?

Reply:
It is like this for consistency with other places doing this